src/lib/sw-bridge.ts

```diff
@@ -40,6 +40,23 @@ async function updateReady(reg: ServiceWorkerRegistration): Promise<void> {
   });
 }
 
+/** Wait for a shared image */
+export function getSharedImage(): Promise<File> {
+  return new Promise((resolve) => {
+    const onmessage = (event: MessageEvent) => {
+      if (event.data.action !== 'load-image') return;
+      resolve(event.data.file);
+      navigator.serviceWorker.removeEventListener('message', onmessage);
+    };
+
+    navigator.serviceWorker.addEventListener('message', onmessage);
+
+    // This message is picked up by the service worker - it's how it knows we're ready to receive
+    // the file.
+    new BroadcastChannel('share-ready').postMessage('share-ready');
```

Comment:
What about iOS?

Reply:
Ah fuck

Reply:
fwiw, `BroadcastChannel` is only used where Share Target V2 is also supported, but I'll make it work using the clients API anyway.